app/Platform/Services/GameListService.php

```diff
@@ -375,7 +377,7 @@ private function getTitleColumn(array $filterOptions): array
         return [
             'header' => 'Title',
             'render' => function ($game) use ($filterOptions) {
-                if (!$filterOptions['console']) {
+                if (isset($filterOptions['console']) && !$filterOptions['console']) {
```

Comment:
I had a slightly different solution to this problem in #2159: https://github.com/RetroAchievements/RAWeb/pull/2159/files#diff-c4abbf0b8b7d6b454423e86a2bb4115262798cf69424713559b116f0ba4c74f3R378

Reply:
I attempted to replace this line with `if (!($filterOptions['console'] ?? false)) {`, however, for some reason it's causing the console labels to reappear:

![Screenshot 2024-01-20 at 10 54 58 AM](https://github.com/RetroAchievements/RAWeb/assets/3984985/ceed672b-e121-4a7b-b7f7-af3d539055de)

Reply:
I agree with your gut feeling.

Addressed in latest. I've added a new public member to `GameListService` called `withSystemNames`, which defaults to `true`.

`SystemController` flips this value to `false`.

I've adjusted `GameListService::getTitleColumn()` to:

```php
private function getTitleColumn(array $filterOptions): array
{
    $consoleName = null;
    if ($this->withConsoleNames) {
        $consoleName = $filterOptions['console'] ?? null;
    }

    return [
        'header' => 'Title',
        'render' => function ($game) use ($consoleName) {
            echo '<td>';
            echo Blade::render('
                <x-game.multiline-avatar
                    :gameId="$ID"
                    :gameTitle="$Title"
                    :gameImageIcon="$ImageIcon"
                    :consoleName="$consoleName"
                />', array_merge($game, ['consoleName' => $consoleName])
            );
            echo '</td>';
        },
    ];
}
```

Reply:
This solution doesn't seem to be working quite right. The console names aren't visible on the hub lists when not grouping by console, and when grouping by console they all just say "1":
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/d6efe4c4-39a2-4b95-9bda-e1cdf6f881f4)

Reply:
Thanks, I think we both noticed this around the same time. Addressed in latest, and further expounded upon here: https://github.com/RetroAchievements/RAWeb/pull/2161#discussion_r1464102596